pkg/kubelet/util/util.go

```diff
@@ -60,3 +60,52 @@ func GetContainerByIndex(containers []v1.Container, statuses []v1.ContainerStatu
 	}
 	return containers[idx], true
 }
+
+type ResourceOpts struct {
+	PodResources       *v1.ResourceRequirements
+	ContainerResources *v1.ResourceRequirements
+}
+
+// GetLimits calculates the resource limits for the container.
+// It uses the container's explicit limits first. If limits for CPU or Memory
+// are not explicitly set at the container level (i.e., they are zero),
+// the corresponding limits from the Pod's resource requirements are applied
+// as defaults to the returned map.
+// TODO(ndixita): Consolidate resource limit logic
+// The implementation of CPU/Memory limit calculation is duplicated here and in
+// pkg/kubelet/kuberuntime/kuberuntime_container_linux.go (getCPULimits/getMemoryLimits).
+// Refactor into a shared utility function.
+func GetLimits(res *ResourceOpts) v1.ResourceList {
```

Comment:
nit: why the ResourceOpts struct, rather than just passing 2 parameters?

Reply:
While testing I swapped the two params :| so for clarity or I could pass container resources as the v1.ResourceRequirements and pod resources as *v1.ResourceRequirements.